apis/addons/v1alpha1/addons_types.go

```diff
@@ -5,18 +5,40 @@ import (
 )
 
 // AddonSpec defines the desired state of Addon.
-type AddonSpec struct{}
+type AddonSpec struct {
+	Namespaces []AddonNamespace `json:"namespaces,omitempty"`
+}
+
+type AddonNamespace struct {
+	Name string `json:"name"`
+}
+
+const (
+	// Available condition indicates that all resources for the Addon are reconciled and healthy
+	Available = "Available"
+)
 
 // AddonStatus defines the observed state of Addon
 type AddonStatus struct {
 	// The most recent generation observed by the controller.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 	// Conditions is a list of status conditions ths object is in.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
-	// Human readable status aggregated from conditions.
-	Phase string `json:"phase,omitempty"`
+	// DEPRECATED: This field is not part of any API contract
+	// it will go away as soon as kubectl can print conditions!
+	// Human readable status - please use .Conditions from code
+	Phase AddonPhase `json:"phase,omitempty"`
 }
 
+type AddonPhase string
+
+// These are the valid phases of an Addon
```

Comment:
```suggestion
// Well-known Addon Phases for printing a Status in kubectl,
// see deprecation notice in AddonStatus for details.
```

Nit: Strictly speaking all strings are allowed in the phase property and this is just a list of well-known phases.

Reply:
Good catch! :+1: